### PR TITLE
Bump to Go 1.22 - Pin latest contrib with Go 1.22

### PR DESCRIPTION
**Pull request by berndverst**
# Description

Pin the latest contrib which now uses Go 1.22, as a result use Go 1.22 in Runtime also.

### CODE REVIEW

**Review by ItalyPaleAle — APPROVED** [review]

**Comment by berndverst** [comment]
FYI - E2E tests have actor related E2E failures, but this isn't related to any contrib changes. Probably should be investigated separately against master branch.

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by berndverst** [comment]
Unclear why this complains about `gotestsum` not being found and as a result fails the unit tests. It works perfectly locally on Apple Silicon.

File: go.mod [hunk 1]
```diff
@@ -177,12 +177,12 @@ require (
...
 	github.com/chebyrash/promise v0.0.0-20230709133807-42ec49ba1459 // indirect
-	github.com/chenzhuoyu/iasm v0.9.0 // indirect
-	github.com/choleraehyq/pid v0.0.17 // indirect
+	github.com/chenzhuoyu/iasm v0.9.1 // indirect
+	github.com/choleraehyq/pid v0.0.18 // indirect
 	github.com/clbanning/mxj/v2 v2.5.6 // indirect
 	github.com/cloudevents/sdk-go/binding/format/protobuf/v2 v2.14.0 // indirect
 	github.com/cloudwego/fastpb v0.0.4-0.20230131074846-6fc453d58b96 // indirect
-	github.com/cloudwego/frugal v0.1.8 // indirect
+	github.com/cloudwego/frugal v0.1.14 // indirect
```
Comment (mikeee) [inline go.mod:185]: Do we need to bump `cloudwego/kitex` and indirectly `cloudwego/frugal`  to a significantly newer minor version upstream in contrib to fix building?

**Comment by berndverst** [comment]
Now tests should run again with CosmosDB 1.0+ (the way a NotFound response was returned changed when the SDK was made stable)

**Comment by berndverst** [comment]
@JoshVanL @yaron2 @artursouza can you merge this?

@JoshVanL by the way it seems that switching the Go version to 1.22 impacts `gotestsum` on macos and other platforms (but not Linux). That's somehow runtime specific.

**Comment by berndverst** [comment]
FYI Windows test is fine - it completed almost all tests but then timed out at the end.

**Comment by JoshVanL** [comment]
Thanks @berndverst! I have opened https://github.com/berndverst/dapr/pull/63 to update the rest of the `go.mod`s and updated an integration test fuzz generator to account for the new Go HTTP route key characters.

**Comment by berndverst** [comment]
> Thanks @berndverst! I have opened [berndverst#63](https://github.com/berndverst/dapr/pull/63) to update the rest of the `go.mod`s and updated an integration test fuzz generator to account for the new Go HTTP route key characters.

I merged your PR @JoshVanL

**Comment by berndverst** [comment]
Feel free to push commits directly to my branch to fix things - as runtime maintainers you can do that. (I do it a lot for contrib contributions to fix up PRs)

**Comment by berndverst** [comment]
Tests are still getting `/bin/sh: gotestsum: command not found` on Mac and Windows. No idea why. I cannot reproduce this locally.

Runtime maintainers, please push commits to fix directly to this PR. I am not looking at PRs opened against my branch.

**Comment by JoshVanL** [comment]
Thanks @berndverst, I have pushed a commit which should resolve the MacOS test runner error.

**Review by yaron2 — APPROVED** [review]

**Comment by marcduiker** [comment]
@holopin-bot @berndverst Enjoy your Dapr badge!

**Pull request merged**